### PR TITLE
Added Docker Support

### PR DESCRIPTION
**Pull request by DarkJ24**
### Features Added

* Added support for running the app as a Docker container by using a `Dockerfile` based on the ruby 2.6.5 image.
* Added `docker-compose.yaml` file to run the generated Docker image with `docker-compose`.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -48,3 +48,6 @@ build-iPhoneSimulator/
 
 # unless supporting rvm < 1.11.0 or doing something fancy, ignore this:
 .rvmrc
+
+# Environment Variables
+.env
```
Comment (sue445) [inline .gitignore]:
```suggestion
.env

```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+WORKDIR /app
+
+# Bundle app source
+COPY . .
+
+RUN bundle install
+
+EXPOSE 3000
+
+CMD ["bundle","exec","puma","-p","3000"]
```
Comment (sue445) [inline Dockerfile]:
```suggestion
CMD ["bundle","exec","puma","-p","3000"]

```

File: docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    working_dir: "/app"
+    environment:
+      - SLACK_WEBHOOK_URL=${SLACK_WEBHOOK_URL}
+      - SLACK_CHANNEL=${SLACK_CHANNEL}
+      - SLACK_USERNAME=${SLACK_USERNAME}
+      - DEBUG_LOGGING=${DEBUG_LOGGING}
+      - MAX_THREADS=${MAX_THREADS}
+      - WEB_CONCURRENCY=${WEB_CONCURRENCY}
+    ports:
+      - "${PORT}:3000"
```
Comment (sue445) [inline docker-compose.yaml]:
```suggestion
      - "${PORT}:3000"
```

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+services:
+
+  docker1.webhook.healthchain.cr:
+    image: soinsalud/dockerhub-slack-webhook:latest
+    container_name: docker1.webhook.healthchain.cr
+    hostname: docker1.webhook.healthchain.cr
+    working_dir: "/app"
+    environment:
+      - SLACK_WEBHOOK_URL=${SLACK_WEBHOOK_URL}
```
Comment (sue445) [inline docker-compose.yaml]:
```suggestion
      - SLACK_WEBHOOK_URL
```

`=${SLACK_WEBHOOK_URL}` is needless

c.f. https://docs.docker.com/compose/environment-variables/#pass-environment-variables-to-containers

File: docker-compose.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+      - SLACK_CHANNEL=${SLACK_CHANNEL}
```
Comment (sue445) [inline docker-compose.yaml]:
```suggestion
      - SLACK_CHANNEL
```

File: docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+      - SLACK_USERNAME=${SLACK_USERNAME}
```
Comment (sue445) [inline docker-compose.yaml]:
```suggestion
      - SLACK_USERNAME
```

File: docker-compose.yaml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+      - DEBUG_LOGGING=${DEBUG_LOGGING}
```
Comment (sue445) [inline docker-compose.yaml]:
```suggestion
      - DEBUG_LOGGING
```

File: docker-compose.yaml [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+      - MAX_THREADS=${MAX_THREADS}
```
Comment (sue445) [inline docker-compose.yaml]:
```suggestion
      - MAX_THREADS
```

File: docker-compose.yaml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+      - WEB_CONCURRENCY=${WEB_CONCURRENCY}
```
Comment (sue445) [inline docker-compose.yaml]:
```suggestion
      - WEB_CONCURRENCY
```

File: README.md [hunk 10]
```diff
@@ -27,3 +27,31 @@ e.g.
...
+SLACK_USERNAME="Docker Hub Build"
+DEBUG_LOGGING=false
+MAX_THREADS=5
+WEB_CONCURRENCY=1
+```
+
+### 3. Run with Docker Compose
+
+```
+docker compose up -d
```
Comment (sue445) [inline README.md]:
```suggestion
docker-compose up -d
```

I think this is typo

File: docker-compose.yaml [hunk 11]
```diff
@@ -0,0 +1,18 @@
+version: '2.4'
+
+services:
+
+  docker1.webhook.healthchain.cr:
+    image: soinsalud/dockerhub-slack-webhook:latest
```
Comment (sue445) [inline docker-compose.yaml]: Why do we need to use soinsalud's dockerhub image?

And `container_name` and `hostname` are needless.

I prefer to followings

```yaml
  app:
    build: .
    working_dir: "/app"
```

File: README.md [hunk 12]
```diff
@@ -27,3 +27,31 @@ e.g.
...
 ### 4. Push image to Docker Hub
+
+## Running as a Docker Container
+
+You can also run this app as a Docker container instead of deploying it to Heroku.
+
+### 1. Generate Docker Image
+
+```
+docker build -t dockerhub-slack-webhook .
```
Comment (sue445) [inline README.md]: This step is needless 

c.f. https://github.com/sue445/dockerhub-slack-webhook/pull/223#discussion_r349905796

**Comment by DarkJ24** [comment]
Added all changes as suggested.

**Pull request merged**